src/components/tl/ucp/allgatherv/allgatherv.c

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+#include "tl_ucp.h"
+#include "allgatherv.h"
+
+ucc_status_t ucc_tl_ucp_allgatherv_ring_start(ucc_coll_task_t *task);
+ucc_status_t ucc_tl_ucp_allgatherv_ring_progress(ucc_coll_task_t *task);
+
+ucc_status_t ucc_tl_ucp_allgatherv_init(ucc_tl_ucp_task_t *task)
+{
+    if ((task->args.src.info.datatype == UCC_DT_USERDEFINED) ||
+        (task->args.dst.info.datatype == UCC_DT_USERDEFINED)) {
```

Comment:
why do we need to check this here? can we check with during ucc_collective_init() init ?

Reply:
The idea is that this particular implementation of allgatherv doesn't support userdefined dataype. Where do you think we should move it? UCX TL coll init, CL basic coll init or core UCC coll init?

Reply:
agree with Sergey, some other TL may support this. The check should be at TL level imo.

Reply:
@vspetrov do you mean check it in ucc_tl_ucp_coll_init instead of in each collective separately?

Reply:
i meant each collective algorithm separately.